alternator/streams.cc

```diff
@@ -841,6 +841,7 @@ future<executor::request_return_type> executor::get_records(client_state& client
                 auto item = rjson::empty_object();
                 describe_single_item(*selection, row, attr_names, item, true);
                 if (!item.ObjectEmpty()) {
+                    describe_single_item(*selection, row, key_names, item);
```

Comment:
You mentioned in the commit message - and also in code here - that you only need to insert the key if there is anything else in the item. But it's possible there is nothing in the item besides the key, and in this case, isn't item an empty object to which you **should** add the key? Did you check this case as well? I'll write a test for it momentarily so I can check your patch myself.

Reply:
Handling all cases turns out requires fixing cdc to not generate preimages for non-existent rows. The generation of which is weird that it came to be in the first place.